service/service.go

```diff
@@ -686,12 +687,65 @@ func (s *Service) MakeNewSecret(ctx context.Context, req *pbsvc.UserRequest) (*p
 }
 
 // VerifyEmailToken checks if received token is found in the email_tokens table.
-// If found and token is NOT expired, returns OK.
-// If found, but token IS expired, the token row with expired token will be deleted,
-// service will generate a new email token and insert into email_tokens table,
-// and return ERROR with token expired message.
-// If token is not found, return ERROR with token does not exist message.
+// If found and token is NOT expired, deletes token row and returns OK.
+// If found, but token IS expired, it will return a expired token error.
+// Additionally for expired tokens, if user is new, it will delete token AND user row, else just deletes the token row.
+// If token is not found, return error with token does not exist message.
 func (s *Service) VerifyEmailToken(ctx context.Context, req *pbsvc.UserRequest) (*pbsvc.UserResponse, error) {
-	// TODO
-	return nil, nil
+	logger.RequestService("VerifyEmailToken")
+
+	if ok := serviceStateLocker.isStateAvailable(); !ok {
+		logger.Error(consts.VerifyEmailToken, consts.ErrServiceUnavailable.Error())
+		return nil, consts.ErrStatusServiceUnavailable
+	}
+
+	if req == nil {
+		return nil, consts.ErrStatusNilRequestUser
```

Comment:
At a minimum, all errors in service or API called externally have to be logged before returning not matter how simple they look like. 

We do NOT have to log this errors in other source code such `db.go` because we have a good design of having the error go back to `service.go`